CalenderView/app/src/main/java/com/non_name_hero/calenderview/calendar/CalendarAdapter.java

```diff
@@ -69,112 +56,86 @@ public int getCount() {
 
         //日付のみ表示させる
-        calendarCellBinding.setDate(PigLeadUtils.formatD.format(cellDate));
+        SimpleDateFormat dateFormat = new SimpleDateFormat("d", Locale.US);
+        //holder.dateTextに日付をセットする。
+        holder.dateText.setText(dateFormat.format(dateArray.get(position)));
+        //mSelectedDate =
 
         //当月以外のセルをグレーアウト
-        if (mDateManager.isCurrentMonth(cellDate)) {
+        if (mDateManager.isCurrentMonth(dateArray.get(position))) {
             //当日の背景を黄色に
-            if (mDateManager.getCurrentDate().equals(cellDate)) {
-                calendarCellBinding.calendarCell.setBackgroundColor(mContext.getResources().getColor(R.color.currentDayColor));
+            if (mDateManager.getCurrentDate().equals(dateArray.get(position))) {
+                convertView.setBackgroundColor(Color.YELLOW);
             } else {
-                calendarCellBinding.calendarCell.setBackgroundColor(Color.WHITE);
+                convertView.setBackgroundColor(Color.WHITE);
             }
         } else {
-            calendarCellBinding.calendarCell.setBackgroundColor(mContext.getResources().getColor(R.color.notCurrentMonth));
+            convertView.setBackgroundColor(Color.LTGRAY);
         }
 
         //祝日、日曜日を赤、土曜日を青に
-        setCalendarCellBase(
-                calendarCellBinding,
-                PigLeadUtils.formatYYYYMMDD.format(cellDate),
-                mDateManager.getDayOfWeek(cellDate)
-        );
-
-        //スケジュールをセルに追記
-        if (mCalendarMap != null) {
-            List<CalendarData> scheduleList = mCalendarMap.get(PigLeadUtils.formatYYYYMMDD.format(cellDate));
-            //スケジュールを追加
-            for (int i = 0; i < 4; i++) {
-                if (mCalendarMap.containsKey(PigLeadUtils.formatYYYYMMDD.format(cellDate)) && i < scheduleList.size()) {
-                    setScheduleText(scheduleList.get(i), calendarCellBinding.scheduleList);
-                } else {
-                    break;
-                }
-            }
-        }
+        judgeHoliday(holder, position, mDateManager.getDayOfWeek(dateArray.get(position)));
+
 
-        //ビューを即時更新
-        calendarCellBinding.executePendingBindings();
-        return calendarCellBinding.getRoot();
+        return convertView;
     }
 
-    /**
-     * カレンダーセルベースを作成
-     */
-    private void setCalendarCellBase(CalendarCellBinding view, String date, int dayOfWeek) {
-        view.scheduleList.removeAllViews();
-        List<CalendarData> schedules = new ArrayList<>();
-        Boolean isHoliday = Boolean.FALSE;
-        if (mHolidayMap != null && mHolidayMap.containsKey(date)) {
-            schedules = mHolidayMap.get(date);
-            for (CalendarData schedule : schedules) {
-                //祝日の場合
-                setScheduleText(schedule, view.scheduleList);
+    //祝日を判定
+    private void judgeHoliday(ViewHolder holder, int position, int day) {
+        //祝日の場合
+        if (mDateManager.getHoliday(dateArray.get(position)) != "") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.